src/main/java/com/project/stockexchangeappbackend/rest/StockController.java

```diff
@@ -180,4 +180,18 @@ public void delete(@ApiParam(value = "The id of stock to delete.", required = tr
         return resourceService.getStockOwners(pageable, specification, id);
     }
 
+    @PatchMapping("/{id}/amount")
+    @PreAuthorize("hasRole('ADMIN')")
+    @ApiOperation(value = "Update amount of given stock", notes = "Required role ADMIN")
+    @ApiResponses({@ApiResponse(code = 200, message = "Stock's amount was successfully updated."),
+            @ApiResponse(code = 400, message = "The request could not be understood or was missing required parameters.",
+                    response = ErrorResponse.class),
+            @ApiResponse(code = 403, message = "Access Denied."),
+            @ApiResponse(code = 404, message = "Given stock not found", response = ErrorResponse.class)})
+    public void updateStockAmount(@ApiParam(value = "The stock's id.", required = true) @PathVariable(name = "id") Long stockId,
+                                  @ApiParam(value = "Owner object", required = true)
+                                  @RequestBody @Valid List<OwnerDTO> ownerDTOList) {
```

Comment:
Where is final amount of updating stock? I think it would be better if you would do like CreateStockDTO
Also I am not sure validation defined this way really works